pennylane/transforms/split_non_commuting.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the tape transform that splits non-commuting terms
+"""
+# pylint: disable=protected-access
+import pennylane as qml
+import numpy as np
+
+from .batch_transform import batch_transform
+
+
+@batch_transform
+def split_non_commuting(tape):
+    r"""
+    Splits a qnode measuring non-commuting observables into groups of commuting observables.
+
+    Args:
+        qnode (pennylane.QNode or .QuantumTape): quantum tape of QNode that contains a list of non-commuting observables to measure.
+
+    Returns:
+        qnode (pennylane.QNode) or tuple[List[.QuantumTape], function]: If a QNode is passed,
+        it returns a QNode capable of handling non-commuting groups.
+        If a tape is passed, returns a tuple containing a list of
+        quantum tapes to be evaluated, and a function to be applied to these
+        tape executions to restore the ordering of the inputs.
+
+    **Example**
+
+    This transform allows us to transform a QNode that measures
+    non-commuting observables to *multiple* circuit executions
+    with qubit-wise commuting groups:
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.transforms.split_non_commuting
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x,wires=0)
+            return [qml.expval(qml.PauliX(0)), qml.expval(qml.PauliZ(0))]
+
+    Instead of decorating the QNode, we can also create a new function that yields the same result in the following way:
+
+    .. code-block:: python3
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x,wires=0)
+            return [qml.expval(qml.PauliX(0)), qml.expval(qml.PauliZ(0))]
+
+        circuit = qml.transforms.split_non_commuting(circuit)
+
+    Internally, the QNode is split into groups of commuting observables when executed:
+
+    >>> print(qml.draw(circuit)(0.5))
+    0: ──RX(0.50)─┤  <X>
+    \
+    0: ──RX(0.50)─┤  <Z>
```

Comment:
I think to avoid needing to include the `\`, you could do this:

```rest
.. code-block:: pycon

    >>> print(qml.draw(circuit)(0.5))
    0: ──RX(0.50)─┤  <X>

    0: ──RX(0.50)─┤  <Z>

Reply:
[but I'm not entirely sure]

Reply:
I tried doing this and it just took the first line as the output and the second one as regular text! I think I also tried a whitespace but then I got a pylint error for trailing whitespaces :D

Reply:
Ah! Weird 🤔 Perhaps we just leave it as is for now then :)